libs/pavex/src/lib.rs

```diff
@@ -229,4 +230,260 @@ pub use pavex_macros::singleton;
 /// [`Blueprint::import`]: crate::blueprint::Blueprint::import
 pub use pavex_macros::transient;
 
+/// Define a [wrapping middleware](https://pavex.dev/docs/guide/middleware/wrapping/).
+///
+/// # Example
+///
+/// A middleware that applies a timeout to all incoming requests:
+///
+/// ```rust
+/// use pavex::middleware::Next;
+/// use pavex::response::Response;
+/// use tokio::time::error::Elapsed;
+///
+/// #[pavex::wrap]
+/// pub async fn timeout<C>(next: Next<C>) -> Result<Response, Elapsed>
+/// where
+///     C: IntoFuture<Output = Response>,
+/// {
+///     let max_duration = std::time::Duration::from_secs(20);
+///     tokio::time::timeout(max_duration, next.into_future()).await
+/// }
+/// ```
+///
+/// # Guide
+///
+/// Check out the ["Middlewares"](https://pavex.dev/docs/guide/middleware/)
+/// section of Pavex's guide for a thorough introduction to middlewares
+/// in Pavex applications.
+///
+/// # Registration
+///
+/// You must invoke [`Blueprint::wrap`] to register the newly-defined middleware
+/// with your [`Blueprint`].
+/// `#[pavex::wrap]` generates a constant that can be used to refer to
+/// the newly-defined middleware when interacting with your [`Blueprint`]:
+///
+/// ```rust
+/// use pavex::blueprint::Blueprint;
+///
+/// # use pavex::middleware::Next;
+/// # use pavex::response::Response;
+/// # use tokio::time::error::Elapsed;
+/// # #[pavex::wrap]
+/// # pub async fn timeout<C>(next: Next<C>) -> Result<Response, Elapsed>
+/// # where
+/// #     C: IntoFuture<Output = Response>,
+/// # {
+/// #     let max_duration = std::time::Duration::from_secs(20);
+/// #     tokio::time::timeout(max_duration, next.into_future()).await
+/// # }
+/// #
+/// # fn main() {
+/// let mut bp = Blueprint::new();
+/// // The generated constant, by default, is named `<fn_name>_ID`,
```

Comment:
The `_ID` is wrong now, isn't?

Reply:
It is indeed, good catch! I'll patch it up as part of #484